code/datums/spells/lichdom.dm

```diff
@@ -89,45 +80,89 @@
 				old_body.dust()
 
-		if(!marked_item) //linking item to the spell
+		if(!linked_phylactery) //becoming a lich
 			message = "<span class='warning'>"
-			for(var/obj/item in hand_items)
-				if(ABSTRACT in item.flags || NODROP in item.flags)
-					continue
-				marked_item = item
-				to_chat(M, "<span class='warning'>You begin to focus your very being into the [item.name]...</span>")
-				break
-
-			if(!marked_item)
-				to_chat(M, "<span class='caution'>You must hold an item you wish to make your phylactery...</span>")
-				return
-
-			spawn(50)
-				if(marked_item.loc != M) //I changed my mind I don't want to put my soul in a cheeseburger!
-					to_chat(M, "<span class='warning'>Your soul snaps back to your body as you drop the [marked_item.name]!</span>")
-					marked_item = null
-					return
+			to_chat(M, "<span class='warning'>You begin the ritual. You feel your heart being pulled out of your chest...</span>")
+			if(do_after(M, 50, target = M))
 				name = "RISE!"
 				desc = "Rise from the dead! You will reform at the location of your phylactery and your old body will crumble away."
 				charge_max = 1800 //3 minute cooldown, if you rise in sight of someone and killed again, you're probably screwed.
 				charge_counter = 1800
 				stat_allowed = 1
-				marked_item.name = "Ensouled [marked_item.name]"
-				marked_item.desc = "A terrible aura surrounds this item, its very existence is offensive to life itself..."
-				marked_item.color = "#003300"
-				to_chat(M, "<span class='userdanger'>With a hideous feeling of emptiness you watch in horrified fascination as skin sloughs off bone! Blood boils, nerves disintegrate, eyes boil in their sockets! As your organs crumble to dust in your fleshless chest you come to terms with your choice. You're a lich!</span>")
+				to_chat(M, "<span class='userdanger'>Clutching your heart in your hand, you watch in horrified fascination as skin sloughs off bone! Blood boils, nerves disintegrate, eyes boil in their sockets! As your organs crumble to dust in your fleshless chest you come to terms with your choice. You're a lich!</span>")
 				current_body = M.mind.current
 				if(ishuman(M))
 					var/mob/living/carbon/human/H = M
 					H.set_species(/datum/species/skeleton)
+					linked_phylactery = new /obj/item/phylactery(get_turf(H))
+					linked_phylactery.linked_lich = H
+					H.put_in_hands(linked_phylactery)
 					H.unEquip(H.wear_suit)
 					H.unEquip(H.head)
 					H.unEquip(H.shoes)
 					H.unEquip(H.head)
 					H.equip_to_slot_or_del(new /obj/item/clothing/suit/wizrobe/black(H), slot_wear_suit)
 					H.equip_to_slot_or_del(new /obj/item/clothing/head/wizard/black(H), slot_head)
 					H.equip_to_slot_or_del(new /obj/item/clothing/shoes/sandal(H), slot_shoes)
-					H.equip_to_slot_or_del(new /obj/item/clothing/under/color/black(H), slot_w_uniform)
\ No newline at end of file
+					H.equip_to_slot_or_del(new /obj/item/clothing/under/color/black(H), slot_w_uniform)
+
+
+/obj/item/phylactery
```

Comment:
I don't really like this, to be honest.

Ensouled Rubber Duckies, toilet water filled beakers, plushies, and what have you are great and allows the lich to use creative things.

Reply:
Honestly agree. It allows creativity and makes the crew have to work harder. Which they should have to do, rather than just looking for a green heart somewhere. Sure you can probably hide it well..but still.

Reply:
I will look into a way to have this work with any item.